src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -2605,21 +2605,6 @@ class RenameStatementSegment(BaseSegment):
     )
 
 
-@tsql_dialect.segment(replace=True)
-class DropStatementSegment(BaseSegment):
-    """A `DROP` statement.
-
-    Overriding ANSI to add optional delimiter.
-    """
-
-    type = "drop_statement"
-    match_grammar = ansi_dialect.get_segment("DropStatementSegment").match_grammar.copy(
-        insert=[
-            Ref("DelimiterSegment", optional=True),
-        ],
-    )
```

Comment:
So this one is slightly different from ANSI, T-SQL has some weirdness with how it handles delimiters (`;`) so we have to insert them to the end of statements.

Think you'll basically need an equivalent one of these classes here for each of the statements we split the ANSI `DropStatementSegment` into

Reply:
@jpy-git  Am I missing something on this one? Tests are passing but admittedly I'm not familiar with Tsql at all. my assumption is that it would be better to not have additional statements in tsql unless they're needed?

Reply:
I don't think there are any tests for these statements in T-SQL then, let me make a quick PR against yours to add these in.

Reply:
addressed in my PR